runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -393,42 +401,54 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
       // Use a sequencing scheme to re-order compilation requests
       //
       clientSession->getSequencingMonitor()->enter();
-      clientSession->updateMaxReceivedSeqNo(seqNo);
-      if (seqNo > clientSession->getExpectedSeqNo()) // Out of order messages
+      clientSession->updateMaxReceivedSeqNo(seqNo); // TODO: why do I need this?
+      // This request can go through as long as criticalSeqNo has been processed
+      if (criticalSeqNo > clientSession->getLastProcessedCriticalSeqNo())
          {
-         // Park this request until the missing ones arrive
+         // Park this request until `criticalSeqNo` arrives and is processed
          if (TR::Options::getVerboseOption(TR_VerboseJITServer))
-            TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "compThreadID=%d out-of-sequence msg detected for clientUID=%llu seqNo=%u > expectedSeqNo=%u. Parking this thread (entry=%p)",
-            getCompThreadId(), (unsigned long long)clientId, seqNo, clientSession->getExpectedSeqNo(), &entry);
-
+            TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "compThreadID=%d out-of-sequence msg=%u detected for clientUID=%llu criticalSeqNo=%u > lastCriticalSeqNo=%u. Parking this thread (entry=%p)",
+            getCompThreadId(), seqNo, (unsigned long long)clientId, criticalSeqNo, clientSession->getLastProcessedCriticalSeqNo(), &entry);
+         // TODO: must change this trace point
          Trc_JITServerOutOfSequenceMessage(compThread, getCompThreadId(), clientSession,
-               (unsigned long long)clientId, &entry, seqNo, clientSession->getExpectedSeqNo(), clientSession->getNumActiveThreads());
+               (unsigned long long)clientId, &entry, seqNo, clientSession->getLastProcessedCriticalSeqNo(), clientSession->getNumActiveThreads());
```

Comment:
Can we put this into an issue that describes how/why it must be changed?

Reply:
I added a new tracepoint to reflect the change in the message I wanted to convey